internal/ethapi/api.go

```diff
@@ -1475,9 +1512,45 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	if receipt.ContractAddress != (common.Address{}) {
 		fields["contractAddress"] = receipt.ContractAddress
 	}
+	if authToken, ok := s.b.SupportsMultitenancy(ctx); ok {
+		extraDataReader, err := s.b.AccountExtraDataStateReaderByNumber(ctx, rpc.BlockNumber(blockNumber))
+		if err != nil {
+			return nil, fmt.Errorf("no account extra data reader at block %v: %w", blockNumber, err)
+		}
+
+		filteredLogs := make([]*types.Log, 0)
+		for _, l := range receipt.Logs {
+			ok, err := s.hasAccessToContract(ctx, authToken, extraDataReader, l.Address)
+			if err != nil {
+				return nil, err
+			}
+			if ok {
+				filteredLogs = append(filteredLogs, l)
+			}
+		}
+		fields["logs"] = filteredLogs
+		receiptClone := &types.Receipt{PostState: receipt.PostState, Status: receipt.Status, Logs: filteredLogs}
+		fields["logsBloom"] = types.CreateBloom(types.Receipts{receiptClone})
+	}
 	return fields, nil
 }
 
+func (s *PublicTransactionPoolAPI) hasAccessToContract(ctx context.Context, authToken *proto.PreAuthenticatedAuthenticationToken, extraDataReader vm.AccountExtraDataStateReader, addr common.Address) (bool, error) {
+	attributes := make([]*multitenancy.ContractSecurityAttribute, 0)
```

Comment:
MINOR: it looks like we are using only one attribute do we need the slice?

Reply:
removed the slice